internal/yunikorn/sync.go

```diff
@@ -260,55 +208,57 @@ func (s *Service) upsertPartitionNodes(ctx context.Context, partitions []*dao.Pa
 	return nil
 }
 
-// upsertApplications fetches applications for each queue and upserts them into the database
-func (s *Service) upsertApplications(ctx context.Context, queues []*dao.PartitionQueueDAOInfo) error {
+// syncApplications fetches applications for each queue and upserts them into the database
+func (s *Service) syncApplications(ctx context.Context) error {
 	logger := log.FromContext(ctx)
 
-	// Create a wait group as a separate goroutine will be spawned for each partition
-	wg := sync.WaitGroup{}
-	wg.Add(len(queues))
-
-	// Protect the applications and errors slices with a mutex so multiple goroutines can safely append queues
-	mutex := sync.Mutex{}
-
-	var errs []error
-	var apps []*dao.ApplicationDAOInfo
+	applications, err := s.client.GetApplications(ctx, "", "")
+	if err != nil {
+		return fmt.Errorf("could not get applications: %v", err)
+	}
```

Comment:
This will only give us the applications in the `default` partition. Should we loop over all the partitions and do this sync logic?

Reply:
All applications are applied to the default partition so it is okay to fetch it like this